### PR TITLE
Fix code style for cpp model api demos

### PR DESCRIPTION
**Pull request by ivikhrev**
Continue of https://github.com/openvinotoolkit/open_model_zoo/pull/3418

### CODE REVIEW

File: demos/classification_benchmark_demo/cpp/grid_mat.hpp [hunk 1]
```diff
@@ -94,21 +101,28 @@ class GridMat {
         cv::Scalar textColor;
         switch (predictionResul) {
             case PredictionResult::Correct:
-                textColor = cv::Scalar(75, 255, 75); break;   // green
+                textColor = cv::Scalar(75, 255, 75);
+                break;  // green
```
Comment (Wovchena) [inline demos/classification_benchmark_demo/cpp/grid_mat.hpp:105]:
```suggestion
                textColor = cv::Scalar(75, 255, 75);  // green
                break;
```

File: demos/classification_benchmark_demo/cpp/grid_mat.hpp [hunk 2]
```diff
@@ -94,21 +101,28 @@ class GridMat {
...
             case PredictionResult::Incorrect:
-                textColor = cv::Scalar(50, 50, 255); break;   // red
+                textColor = cv::Scalar(50, 50, 255);
+                break;  // red
```
Comment (Wovchena) [inline demos/classification_benchmark_demo/cpp/grid_mat.hpp:108]:
```suggestion
                textColor = cv::Scalar(50, 50, 255);  // red
                break;
```

File: demos/classification_benchmark_demo/cpp/grid_mat.hpp [hunk 3]
```diff
@@ -94,21 +101,28 @@ class GridMat {
...
             case PredictionResult::Unknown:
-                textColor = cv::Scalar(200, 10, 10); break;  // blue
+                textColor = cv::Scalar(200, 10, 10);
+                break;  // blue
```
Comment (Wovchena) [inline demos/classification_benchmark_demo/cpp/grid_mat.hpp:111]:
```suggestion
                textColor = cv::Scalar(200, 10, 10);  // blue
                break;
```

File: demos/classification_benchmark_demo/cpp/grid_mat.hpp [hunk 4]
```diff
@@ -129,11 +143,13 @@ class GridMat {
...
+    static constexpr const char testMessage[] = "Testing, please wait...";
     int baseline;
     cv::Size textSize;
     cv::Size accuracyMessageSize;
     cv::Size testMessageSize;
 };
 
-const std::string GridMat::testMessage = "Testing, please wait...";
+// defenition of the static member needed to remove clang build error
+// https://stackoverflow.com/questions/8016780/undefined-reference-to-static-constexpr-char
```
Comment (Wovchena) [inline demos/classification_benchmark_demo/cpp/grid_mat.hpp:154]:
```suggestion
```
I don't think you need to explain this. That's how the language works.

File: demos/common/cpp/models/src/detection_model_faceboxes.cpp [hunk 5]
```diff
@@ -198,26 +215,32 @@ std::vector<ModelFaceBoxes::Anchor> filterBoxes(const ov::Tensor& boxesTensor, c
...
+                         static_cast<float>(predCtrY + 0.5f * predH)});
     }
 
     return boxes;
 }
 
 std::unique_ptr<ResultBase> ModelFaceBoxes::postprocess(InferenceResult& infResult) {
-    // --------------------------- Filter scores and get valid indices for bounding boxes----------------------------------
+    // --------------------------- Filter scores and get valid indices for bounding
+    // boxes----------------------------------
```
Comment (Wovchena) [inline demos/common/cpp/models/src/detection_model_faceboxes.cpp:229]:
```suggestion
    // Filter scores and get valid indices for bounding boxes
```

File: demos/common/cpp/models/src/detection_model_faceboxes.cpp [hunk 6]
```diff
@@ -198,26 +215,32 @@ std::vector<ModelFaceBoxes::Anchor> filterBoxes(const ov::Tensor& boxesTensor, c
...
     const auto scoresTensor = infResult.outputsData[outputsNames[1]];
     const auto scores = filterScores(scoresTensor, confidenceThreshold);
 
-    // --------------------------- Filter bounding boxes on indices -------------------------------------------------------
+    // --------------------------- Filter bounding boxes on indices
+    // -------------------------------------------------------
```
Comment (Wovchena) [inline demos/common/cpp/models/src/detection_model_faceboxes.cpp:234]:
```suggestion
    // Filter bounding boxes on indices
```

File: demos/common/cpp/models/src/detection_model_faceboxes.cpp [hunk 7]
```diff
@@ -198,26 +215,32 @@ std::vector<ModelFaceBoxes::Anchor> filterBoxes(const ov::Tensor& boxesTensor, c
...
     auto boxesTensor = infResult.outputsData[outputsNames[0]];
     std::vector<Anchor> boxes = filterBoxes(boxesTensor, anchors, scores.first, variance);
 
-    // --------------------------- Apply Non-maximum Suppression ----------------------------------------------------------
+    // --------------------------- Apply Non-maximum Suppression
+    // ----------------------------------------------------------
```
Comment (Wovchena) [inline demos/common/cpp/models/src/detection_model_faceboxes.cpp:239]:
```suggestion
    // Apply Non-maximum Suppression
```

File: demos/common/cpp/models/src/detection_model_faceboxes.cpp [hunk 8]
```diff
@@ -198,26 +215,32 @@ std::vector<ModelFaceBoxes::Anchor> filterBoxes(const ov::Tensor& boxesTensor, c
...
     const std::vector<int> keep = nms(boxes, scores.second, boxIOUThreshold);
 
-    // --------------------------- Create detection result objects --------------------------------------------------------
+    // --------------------------- Create detection result objects
+    // --------------------------------------------------------
```
Comment (Wovchena) [inline demos/common/cpp/models/src/detection_model_faceboxes.cpp:243]:
```suggestion
    // Create detection result objects
```

**Review by Wovchena — CHANGES_REQUESTED** [review]

File: demos/classification_benchmark_demo/cpp/grid_mat.hpp (hunk 4 above)
Comment (ivikhrev) [reply demos/classification_benchmark_demo/cpp/grid_mat.hpp:154]: gcc for example is fine without this definition, so I thought this is clang behavior only.

**Comment by ivikhrev** [comment]
@jkamelin  python precommit fails. Is it connected with https://github.com/openvinotoolkit/open_model_zoo/pull/3424 ?
```
20:15:39  + ./venv/bin/omz_info_dumper --list src/demos/3d_segmentation_demo/python/models.lst
20:15:39  Traceback (most recent call last):
20:15:39    File "./venv/bin/omz_info_dumper", line 5, in <module>
20:15:39      from openvino.model_zoo.omz_info_dumper import main
20:15:39    File "/home/jenkins/workspace/omz/precommit/python_demos_ubuntu18/venv/lib/python3.6/site-packages/openvino/model_zoo/omz_info_dumper.py", line 21, in <module>
20:15:39      from openvino.model_zoo import _configuration, _common
20:15:39    File "/home/jenkins/workspace/omz/precommit/python_demos_ubuntu18/venv/lib/python3.6/site-packages/openvino/model_zoo/_configuration.py", line 23, in <module>
20:15:39      from openvino.model_zoo import _common
20:15:39    File "/home/jenkins/workspace/omz/precommit/python_demos_ubuntu18/venv/lib/python3.6/site-packages/openvino/model_zoo/_common.py", line 23, in <module>
20:15:39      from . import __version__
20:15:39  ImportError: cannot import name '__version__'
```

File: demos/classification_benchmark_demo/cpp/grid_mat.hpp (hunk 4 above)
Comment (Wovchena) [reply demos/classification_benchmark_demo/cpp/grid_mat.hpp:154]: It could be some kind of compiler extension enabled by default. But ok

Comment (ivikhrev) [reply demos/classification_benchmark_demo/cpp/grid_mat.hpp]: I already removed)

Comment (Wovchena) [reply demos/classification_benchmark_demo/cpp/grid_mat.hpp]: Whatever

**Comment by jkamelin** [comment]
> @jkamelin python precommit fails. Is it connected with #3424 ?
> 
> ```
> 20:15:39  + ./venv/bin/omz_info_dumper --list src/demos/3d_segmentation_demo/python/models.lst
> 20:15:39  Traceback (most recent call last):
> 20:15:39    File "./venv/bin/omz_info_dumper", line 5, in <module>
> 20:15:39      from openvino.model_zoo.omz_info_dumper import main
> 20:15:39    File "/home/jenkins/workspace/omz/precommit/python_demos_ubuntu18/venv/lib/python3.6/site-packages/openvino/model_zoo/omz_info_dumper.py", line 21, in <module>
> 20:15:39      from openvino.model_zoo import _configuration, _common
> 20:15:39    File "/home/jenkins/workspace/omz/precommit/python_demos_ubuntu18/venv/lib/python3.6/site-packages/openvino/model_zoo/_configuration.py", line 23, in <module>
> 20:15:39      from openvino.model_zoo import _common
> 20:15:39    File "/home/jenkins/workspace/omz/precommit/python_demos_ubuntu18/venv/lib/python3.6/site-packages/openvino/model_zoo/_common.py", line 23, in <module>
> 20:15:39      from . import __version__
> 20:15:39  ImportError: cannot import name '__version__'
> ```

Fixed with https://github.com/openvinotoolkit/open_model_zoo/pull/3436

**Comment by ivikhrev** [comment]
Jenkins please retry a build

**Review by Wovchena — APPROVED** [review]

**Pull request merged**